src/components/Form.tsx

```diff
@@ -66,54 +68,50 @@ const TextAsker = ({
 /**
  * @summary Renders Form from question[]
  * @param questions questions to render into components
- * @param initResponses mapped type from question id to previously saved response
- * @param onChange Function that takes in responses json and pushes it to the db
- * @param id The id that will be given to the form HTML element. This allows linking external submit buttons.
+ * @param responses mapped type from question id to previously saved response
+ * @param onChange Invoked at every edit on form
+ * @param onAutosave Invoked after user edits the form and pauses for `waitTime` milliseconds.
+ * @param waitTime Autosave debounce wait time in milliseconds
  */
 const Form = ({
   questions,
-  initResponses = {},
-  onSubmit,
-  id,
+  responses = {},
+  onChange = () => {},
+  onAutosave = () => {},
+  waitTime = 3000,
```

Comment:
rename to `autosaveInterval`? `waitTime` seems a little general and the connection to `onAutosave` is not clear.